source/Nevermore.Tests/QueryBuilderFixture/QueryBuilderStateFixture.cs

```diff
@@ -176,6 +199,70 @@ public void ShouldNotModifyQueryBuilderStateFromToListPageinatedWithCount()
 ORDER BY [Id]");
         }
 
+        [Test]
+        public void ShouldNotModifyQueryBuilderStateFromToListPaginatedWithQueryHint()
+        {
+            var queryBuilder = QueryBuilder("Accounts")
+                .Option("OPTIMIZE FOR UNKNOWN")
+                .Option("OPTIMIZE FOR UNKNOWN") // make sure it will deduplicate
+                .Option("FAST 1");
+
+            queryBuilder.ToList(10, 20);
+
+            LastExecutedQuery().ShouldBeEquivalentTo(@"SELECT *
+FROM (
+    SELECT *,
+    ROW_NUMBER() OVER (ORDER BY [Id]) AS RowNum
+    FROM [dbo].[Accounts]
+) ALIAS_GENERATED_1
+WHERE ([RowNum] >= @_minrow)
+AND ([RowNum] <= @_maxrow)
+ORDER BY [RowNum]
+OPTION (OPTIMIZE FOR UNKNOWN, FAST 1)");
+
+            queryBuilder.ParameterValues.Count.ShouldBeEquivalentTo(0);
+            queryBuilder.ToList();
+
+            LastExecutedQuery().ShouldBeEquivalentTo(@"SELECT *
+FROM [dbo].[Accounts]
+ORDER BY [Id]
+OPTION (OPTIMIZE FOR UNKNOWN, FAST 1)");
+        }
+
+        [Test]
+        public void ShouldNotModifyQueryBuilderStateFromToListPaginatedWithCountAndQueryHint()
+        {
+            var queryBuilder = QueryBuilder("Accounts")
+                .Option("OPTIMIZE FOR UNKNOWN");
+
+            queryBuilder.ToList(10, 20, out _);
+
+            executedQueries.Count.ShouldBeEquivalentTo(2);
+            executedQueries.First().ShouldBeEquivalentTo(@"SELECT COUNT(*)
+FROM [dbo].[Accounts]
+OPTION (OPTIMIZE FOR UNKNOWN)");
+
+            LastExecutedQuery().ShouldBeEquivalentTo(@"SELECT *
+FROM (
+    SELECT *,
+    ROW_NUMBER() OVER (ORDER BY [Id]) AS RowNum
+    FROM [dbo].[Accounts]
+) ALIAS_GENERATED_1
+WHERE ([RowNum] >= @_minrow)
+AND ([RowNum] <= @_maxrow)
+ORDER BY [RowNum]
+OPTION (OPTIMIZE FOR UNKNOWN)");
+
+            queryBuilder.ParameterValues.Count.ShouldBeEquivalentTo(0);
+
+            queryBuilder.ToList();
+
+            LastExecutedQuery().ShouldBeEquivalentTo(@"SELECT *
+FROM [dbo].[Accounts]
+ORDER BY [Id]
+OPTION (OPTIMIZE FOR UNKNOWN)");
+        }
```

Comment:
My [previous PR](https://github.com/OctopusDeploy/Nevermore/pull/177) would fail to generate a valid SQL query when calling `ToList(int skip, int take)`. I created this PR to fix that.